### PR TITLE
Logout User

### PR DESCRIPTION
**Pull request by SukanyaSen530**
Functionalities -
- User can log out
- User can change the completion status of a task

Extra -
- Added empty, error and loading states for task list page
- Disabled state for status completion checkboxes

### CODE REVIEW

File: src/services/tasks.js [hunk 1]
```diff
@@ -104,8 +104,14 @@ export const deleteTask = async (id, dispatch) => {
 export const toggleTaskCompletion = async (id, dispatch) => {
   const config = getConfig();
 
+  console.log(config);
```
Comment (Prabhav-Sharma) [inline src/services/tasks.js]: Suggestion: Can remove console calls in production unless important for future development

**Review by Prabhav-Sharma — APPROVED** [review]
Clean code and easy to navigate
Liked the naming convention!

File: src/services/tasks.js (hunk 1 above)
Comment (SukanyaSen530) [reply src/services/tasks.js]: thanks for pointing out, will delete it

**Review by JowelTisso — APPROVED** [review]
* Clean code
* Looks Good

**Review by Mohitkumar33 — APPROVED** [review]
Code looks clean

**Pull request merged**